python_modules/libraries/dagster-celery/dagster_celery/executor_docker.py

```diff
@@ -0,0 +1,184 @@
+    (See https://docs.celeryproject.org/en/latest/userguide/configuration.html for details.)
+
+    The executor also exposes the ``broker``, `backend`, and ``include`` arguments to the
+    :py:class:`celery.Celery` constructor.
+
+    In the most common case, you may want to modify the ``broker`` and ``backend`` (e.g., to use
+    Redis instead of RabbitMQ). We expect that ``config_source`` will be less frequently
+    modified, but that when solid executions are especially fast or slow, or when there are
+    different requirements around idempotence or retry, it may make sense to execute pipelines
+    with variations on these settings.
+
+    If you'd like to configure a Celery Docker executor in addition to the
+    :py:class:`~dagster.default_executors`, you should add it to the ``executor_defs`` defined on a
+    :py:class:`~dagster.ModeDefinition` as follows:
+
+    .. code-block:: python
+
+        from dagster import ModeDefinition, default_executors, pipeline
+        from dagster_celery.executor_docker import celery_docker_executor
+
+        @pipeline(mode_defs=[
+            ModeDefinition(executor_defs=default_executors + [celery_docker_executor])
+        ])
+        def celery_enabled_pipeline():
+            pass
+
+    Then you can configure the executor as follows:
+
+    .. code-block:: YAML
+
+        execution:
+          celery-docker:
+            config:
+              docker_image: 'my_repo.com/image_name:latest'
+              broker: 'pyamqp://guest@localhost//'  # Optional[str]: The URL of the Celery broker
+              backend: 'rpc://' # Optional[str]: The URL of the Celery results backend
+              include: ['my_module'] # Optional[List[str]]: Modules every worker should import
+              config_source: # Dict[str, Any]: Any additional parameters to pass to the
+                  #...       # Celery workers. This dict will be passed as the `config_source`
+                  #...       # argument of celery.Celery().
+
+    Note that the YAML you provide here must align with the configuration with which the Celery
+    workers on which you hope to run were started. If, for example, you point the executor at a
+    different broker than the one your workers are listening to, the workers will never be able to
+    pick up tasks for execution.
+
+    If you want to use your private docker registry you have to pass your registry credentials
+    through ENV as follows:
+
+    .. code-block:: YAML
+
+        DOCKER_USERNAME: my_user
+        DOCKER_PASSWORD: my_secret_pass
+        DOCKER_REGISTRY: my_repo.com
+
+    '''
+    check_cross_process_constraints(init_context)
+
+    run_launcher = init_context.instance.run_launcher
+    exc_cfg = init_context.executor_config
+
+    docker_creds = {
+        'username': os.environ.get('DOCKER_USERNAME'),
+        'password': os.environ.get('DOCKER_PASSWORD'),
+        'registry': os.environ.get('DOCKER_REGISTRY'),
+    }
```

Comment:
I think this should be `StringSource` config entries instead of hardcoded environment variables